src/State/Hub.php

```diff
@@ -241,18 +263,37 @@ public function startTransaction(TransactionContext $context): Transaction
     }
 
     /**
-     * Gets the scope bound to the top of the stack.
+     * {@inheritdoc}
+     *
+     * @psalm-suppress MoreSpecificReturnType
+     * @psalm-suppress LessSpecificReturnStatement
      */
-    private function getScope(): Scope
+    public function getTransaction(): ?Transaction
     {
-        return $this->getStackTop()->getScope();
+        $span = $this->span;
+        if (null !== $span && null !== $span->spanRecorder) {
+            /** @phpstan-ignore-next-line */
```

Comment:
All those ignores are hiding a real problem, it may be empty!

Reply:
Fixed in `3.x` thx @Jean85